gef-ui/server/controllers/amendments/what-needs-to-change/post-what-needs-to-change.test.ts

```diff
@@ -0,0 +1,248 @@
+      amendmentOptions,
+    },
+  });
+
+const mockDeal = { exporter: { companyName }, submissionType: DEAL_SUBMISSION_TYPE.AIN, status: DEAL_STATUS.UKEF_ACKNOWLEDGED } as Deal;
+
+const mockFacility = {
+  hasBeenIssued: true,
+} as Facility;
+
+describe('postWhatNeedsToChange', () => {
+  let amendment: PortalFacilityAmendmentWithUkefId;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    jest.spyOn(dtfsCommon, 'isPortalFacilityAmendmentsFeatureFlagEnabled').mockReturnValue(true);
+
+    amendment = new PortalFacilityAmendmentWithUkefIdMockBuilder().withDealId(dealId).withFacilityId(facilityId).withAmendmentId(amendmentId).build();
+
+    getApplicationMock.mockResolvedValue(mockDeal);
+    getFacilityMock.mockResolvedValue({ details: mockFacility });
+    updateAmendmentMock.mockResolvedValue(amendment);
+  });
+
+  afterAll(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should call getApplication with the correct dealId and userToken', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks();
+
+    // Act
+    await postWhatNeedsToChange(req, res);
+
+    // Assert
+    expect(getApplicationMock).toHaveBeenCalledTimes(1);
+    expect(getApplicationMock).toHaveBeenCalledWith({ dealId, userToken });
+  });
+
+  it('should call getFacility with the correct facilityId and userToken', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks();
+
+    // Act
+    await postWhatNeedsToChange(req, res);
+
+    // Assert
+    expect(getFacilityMock).toHaveBeenCalledTimes(1);
+    expect(getFacilityMock).toHaveBeenCalledWith({ facilityId, userToken: req.session.userToken });
+  });
+
+  it('should not call updateAmendment if neither of changeCoverEndDate or changeFacilityValue are true', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks([]);
+
+    // Act
+    await postWhatNeedsToChange(req, res);
+
+    // Assert
+    expect(updateAmendmentMock).toHaveBeenCalledTimes(0);
+  });
+
+  it('should render the page with validation errors if an invalid selection has been made', async () => {
+    // Arrange
+    const changeCoverEndDate = false;
+    const changeFacilityValue = false;
+    const { req, res } = getHttpMocks([]);
+
+    // Act
+    await postWhatNeedsToChange(req, res);
+
+    // Assert
+    const expectedErrors = validationErrorHandler(validateWhatNeedsToChange({ changeCoverEndDate, changeFacilityValue }) as ValidationError);
+
+    const expectedRenderData: WhatNeedsToChangeViewModel = {
+      exporterName: companyName,
+      previousPage: `/gef/application-details/${dealId}`,
+      amendmentFormEmail: STB_PIM_EMAIL,
+      changeCoverEndDate,
+      changeFacilityValue,
+      errors: expectedErrors,
+    };
+
+    expect(res._getStatusCode()).toEqual(HttpStatusCode.Ok);
+    expect(res._getRenderView()).toEqual('partials/amendments/what-needs-to-change.njk');
+    expect(res._getRenderData()).toEqual(expectedRenderData);
+    expect(console.error).toHaveBeenCalledTimes(0);
```

Comment:
Console.error text please.

Reply:
In the case you've commented on it's testing it hasn't been called, so no text test needed. In the final test of the file I check the text.